server/src/test/java/io/deephaven/server/console/ConsoleServiceTest.java

```diff
@@ -0,0 +1,99 @@
+package io.deephaven.server.console;
+
+import io.deephaven.io.log.LogLevel;
+import io.deephaven.io.logger.LogBufferRecord;
+import io.deephaven.proto.backplane.script.grpc.LogSubscriptionData;
+import io.deephaven.proto.backplane.script.grpc.LogSubscriptionRequest;
+import io.deephaven.server.runner.DeephavenApiServerSingleAuthenticatedBase;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.ClientCallStreamObserver;
+import io.grpc.stub.ClientResponseObserver;
+import org.junit.Test;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ConsoleServiceTest extends DeephavenApiServerSingleAuthenticatedBase {
+
+    private static class Observer implements ClientResponseObserver<LogSubscriptionRequest, LogSubscriptionData> {
+        private final CountDownLatch latch;
+        private ClientCallStreamObserver<?> stream;
+        private Throwable error;
+        private CountDownLatch done;
+        private int count;
+
+        public Observer(int expected) {
+            latch = new CountDownLatch(expected);
+            done = new CountDownLatch(1);
+        }
+
+        @Override
+        public void beforeStart(ClientCallStreamObserver<LogSubscriptionRequest> requestStream) {
+            this.stream = requestStream;
+        }
+
+        @Override
+        public void onNext(LogSubscriptionData value) {
+            latch.countDown();
+            ++count;
```

Comment:
This is likely to be invoked from another thread. You need to make `count`, and `error`, `volatile` to avoid a racy test. (Or synchronize `onNext` / `onError`.)

Reply:
I think the semantics provided by the CountDownLatch are good enough - I can refactor a bit to get rid of `count`; and, `onError` / `onComplete` are guaranteed to only be called once, and visibility of `error` should be guarded by `done`. I'll re-push, let me know what you think.